linera-core/src/unit_tests/client_tests.rs

```diff
@@ -1947,10 +1945,8 @@ where
         .block()
         .unwrap()
         .operations
-        .contains(&Operation::System(SystemOperation::Transfer {
-            owner: None,
-            recipient: Recipient::Burn,
-            amount: Amount::from_tokens(1),
+        .contains(&Operation::System(SystemOperation::PublishDataBlob {
+            blob_hash: blob4.id().hash
```

Comment:
Without the new fix, here the `operations` would NOT contain the `PublishDataBlob { blob4_id }` ?

Reply:
No, it would fail before this line. This assertion here is only afterwards, to confirm that both blocks were committed. I changed the "burn" into a "publish data blob" to trigger the bug.

The bug would happen earlier, at the `publish_data_blob` call:

First the client is forced to finalize the conflicting block (existing validated block certificate), so another block gets committed at this height.

Then it would forget the pending blobs, retry its own `PublishDataBlob` operation, and fail.